src/main/java/org/elasticsearch/indices/cache/query/IndicesQueryCache.java

```diff
@@ -266,7 +266,6 @@ public BytesReference call() throws Exception {
             // for now, keep the paged data structure, which might have unused bytes to fill a page, but better to keep
             // the memory properly paged instead of having varied sized bytes
             BytesReference value = out.bytes();
-            assert verifyCacheSerializationSameAsQueryResult(value, context, context.queryResult());
```

Comment:
The assertion checks that BytesQuerySearchResult serializes the same thing as the actual QuerySearchResult, but there is no BytesQuerySearchResult anymore.